cmd/tea/gen_id.go

```diff
@@ -128,6 +128,30 @@ func (i *{{ . }}) UnmarshalText(data []byte) error {
 func (i *{{ . }}) UnmarshalJSON(data []byte) error {
 	return unmarshalUUID((*uuid.UUID)(i), "{{ . }}", data)
 }
+
+func (i *{{ . }}) Scan(src any) error {
+	if src == nil {
+		*i = {{ . }}(uuid.Nil)
+		return nil
+	}
+	switch v := src.(type) {
+	case string:
+		u, err := uuid.Parse(v)
+		if err != nil {
+			return err
+		}
+		*i = {{ . }}(u)
+		return nil
+	case []byte:
+		u, err := uuid.ParseBytes(v)
+		if err != nil {
+			return err
+		}
+		*i = {{ . }}(u)
+		return nil
+	}
+	return fmt.Errorf("unable to scan %T", src)
+}
```

Comment:
The null check can be inside a type switch

Also the `uuid.Scan` looks very different in the `[]byte` case:
```
// Scan implements sql.Scanner so UUIDs can be read from databases transparently.
// Currently, database types that map to string and []byte are supported. Please
// consult database-specific driver documentation for matching types.
func (uuid *UUID) Scan(src interface{}) error {
	switch src := src.(type) {
	case nil:
		return nil

	case string:
		// if an empty UUID comes from a table, we return a null UUID
		if src == "" {
			return nil
		}

		// see Parse for required string format
		u, err := Parse(src)
		if err != nil {
			return fmt.Errorf("Scan: %v", err)
		}

		*uuid = u

	case []byte:
		// if an empty UUID comes from a table, we return a null UUID
		if len(src) == 0 {
			return nil
		}

		// assumes a simple slice of bytes if 16 bytes
		// otherwise attempts to parse
		if len(src) != 16 {
			return uuid.Scan(string(src))
		}
		copy((*uuid)[:], src)

	default:
		return fmt.Errorf("Scan: unable to scan type %T into UUID", src)
	}

	return nil
}
```